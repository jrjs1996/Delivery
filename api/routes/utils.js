const multer = require('multer');

const storage = multer.diskStorage({
  destination: (req, file, cb) => {
    console.log(file);
    cb(null, './uploads/');
  },
  filename: (req, file, cb) => {
    console.log('id', req.body.id);
    cb(null, req.body.id);
  },
});

const fileFilter = (req, file, cb) => {
  if (file.mimetype === 'image/jpeg' || file.mimetype === 'image/png') {
    cb(null, true);
  }
  cb(null, false);
};

module.exports.upload = multer({ storage, fileFilter });
