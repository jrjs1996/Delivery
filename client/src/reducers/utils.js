/**
 *  Converts an array of items to an object with the
 * items properties and the idProperty as keys.
 * @param {Array[object]} itemsArray Array of items.
 * @param {string} idProperty Property to use as key.
 */
export const createItems = (itemsArray, idProperty) => {
  const items = {};
  itemsArray.forEach((item) => {
    items[item[idProperty]] = item;
  });
  return items;
};

/**
 * Inserts an object into state.
 * @param {Object} items Current items in state
 * @param {Object} newItem Item to insert into state
 * @param {String} idProperty Property to use as the key in state.
 */
export const insertItem = (items, newItem, idProperty) => ({
  ...items,
  [newItem[idProperty]]: newItem,
});

/**
 * Updates an object in state.
 * @param {Object} items Current items in state.
 * @param {Object} newItem Item to update in state.
 * @param {String} idProperty Property to use as key to find object.
 */
export const updateItem = (items, newItem, idProperty) => ({
  ...items,
  [newItem[idProperty]]: Object.assign(items[newItem[idProperty]], newItem),
});

/**
 * Deletes an item in state.
 * @param {Object} items Current items in state
 * @param {Object} newItem Item to delete from state.
 * @param {String} idProperty Property to use as key to find object.
 */
export const deleteItem = (items, newItem, idProperty) => {
  const { [newItem[idProperty]]: omit, ...rest } = items;
  return rest;
};
