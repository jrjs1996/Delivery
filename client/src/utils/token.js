export const getToken = () => localStorage.getItem('jwt-token');
export const getIsAdmin = () => localStorage.getItem('isAdmin');
// TODO: Add docs for these functions

/**
 * Save the token and isAdmin in local storage.
 */
export const saveToken = (token, isAdmin) => {
  if (!token || (undefined === isAdmin)) throw new Error('Token and isAdmin Required');

  localStorage.setItem('jwt-token', token);
  localStorage.setItem('isAdmin', isAdmin);
};

/** Removes the token and isAdmin from local storage */
export const removeToken = () => {
  localStorage.removeItem('jwt-token');
  localStorage.removeItem('isAdmin');
};

/**
 * Returns token
 */
export const getTokenInfo = () => {
  try {
    const token = getToken();
    if (token) {
      const payload = window.atob(token.split('.')[1]);
      return JSON.parse(payload);
    }
  } catch (error) {
    console.error(error);
    throw error;
  }
};

/**
 * Returns an object containing the token info and a boolean
 * that is true if the user is an admin, false otherwise.
 * { tokenInfo: *token* , isAdmin *true/false* }
 * Returns false if the user is not authenticated.
 */
export const isAuthed = () => {
  const tokenInfo = getTokenInfo();
  const isAdmin = getIsAdmin();
  if (tokenInfo && tokenInfo.exp > Math.round(new Date() / 1000)) {
    return {
      tokenInfo,
      isAdmin,
    };
  }
  return false;
};

/**
 * Sets authorziation header with the current users token.
 */
export const setAuthHeader = () => ({
  headers: { authorization: `Bearer ${getToken()}` },
});
