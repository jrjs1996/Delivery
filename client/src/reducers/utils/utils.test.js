import * as utils from './utils';

const array = [{ id: 1, value: 'one' }, { id: 2, value: 'two' }];
const state = () => ({
  1: {
    id: 1,
    value: 'one',
  },
  2: {
    id: 2,
    value: 'two',
  },
});

describe('Create items', () => {
  it(' Converts an array of items to an object with the items properties and the idProperty as keys.', () => {
    expect(utils.createItems(array, 'id')).toMatchSnapshot();
  });
});

describe('Insert Item', () => {
  it('Insert an object into state with the key of the object being the value of the key property.', () => {
    expect(utils.insertItem(state(), { id: 4, value: 'four' }, 'id')).toMatchSnapshot();
  });
});

describe('Update Item', () => {
  it('Updates item with matching id property', () => {
    expect(utils.updateItem(state(), { id: 2, value: 'Newtwo' }, 'id')).toMatchSnapshot();
  });
});

describe('Delete Item', () => {
  it('Deletes item with matching id property', () => {
    expect(utils.deleteItem(state(), { id: 2 }, 'id')).toMatchSnapshot();
  });
})
