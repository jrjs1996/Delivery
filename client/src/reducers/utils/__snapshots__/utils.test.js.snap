// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create items  Converts an array of items to an object with the items properties and the idProperty as keys. 1`] = `
Object {
  "1": Object {
    "id": 1,
    "value": "one",
  },
  "2": Object {
    "id": 2,
    "value": "two",
  },
}
`;

exports[`Delete Item Deletes item with matching id property 1`] = `
Object {
  "1": Object {
    "id": 1,
    "value": "one",
  },
}
`;

exports[`Insert Item Insert an object into state with the key of the object being the value of the key property. 1`] = `
Object {
  "1": Object {
    "id": 1,
    "value": "one",
  },
  "2": Object {
    "id": 2,
    "value": "two",
  },
  "4": Object {
    "id": 4,
    "value": "four",
  },
}
`;

exports[`Update Item Updates item with matching id property 1`] = `
Object {
  "1": Object {
    "id": 1,
    "value": "one",
  },
  "2": Object {
    "id": 2,
    "value": "Newtwo",
  },
}
`;
